lib/PackageEnvironment.js

```diff
@@ -184,14 +184,19 @@ function builtInsPerPackage(
   }
   return {
     [`${prefix}__name`]: builtIn(
-      packageJson.name
+      name
     ),
     [`${prefix}__version`]: builtIn(
-      packageJson.version || null
+      version || null
     ),
-    [`${prefix}__root`]: builtIn(
+    [`${prefix}__source_root`]: builtIn(
```

Comment:
New env var here — `$*__source_root` — the real source root, even when package can't be built out of source.